### DISCUSSION TITLE
wezterm asking for calendar, reminders, other access permissions on OSX

### DISCUSSION THREAD

**Discussion by gschwim** [wezterm/wezterm#3943] [Q&A] [answered]
I just started using wezterm this week. I installed on OSX using homebrew, version is `20230408-112425-69ae8472`. I've been using it for 2-3 days straight, and now I am getting popups requesting that it be granted access to Calendar, Photos, Reminders, and other things.  The only thing I'm doing at this time is working in neovim in various local code repos.

Why would a terminal need access to these things? Is this expected behavior for wezterm?

Just as I'm typing this it asked for notifications permission. That it is doing this only after several days of use seems strange.

**Comment by nekowinston** [comment]
The notification permission is for [`toast_notification`](https://wezfurlong.org/wezterm/config/lua/window/toast_notification.html), and Calendar, Reminders, permissions, may be triggered by changing your current directory to `~/Documents` & `~/Photos`, (if you're using iCloud sync) `~/Library`, etc.

It's not unusual for Terminals to request even Full Disk Access, which essentially grants all these at once, since you're effectively not giving the terminal emulator the permissions, but the shell you're running.

Edit: your notification permission request was probably triggered by the [automatic update check](https://wezfurlong.org/wezterm/config/lua/config/check_for_updates.html).

**Reply by wez** [reply]
FWIW, the update check is not one of the things that macOS will warn about.

**Reply by nekowinston** [reply]
Oh yeah that's right, I've had the update check suppressed for so long, that I forgot about it showing up in the terminal window rather than a notification.

I guess it could've been a missing font symbol notification(?)

**Comment by wez** [comment, marked as answer]
How macOS works is that it attributes those permission requests to the ancestor process that has an app bundle.  So if in your shell, which was launched by wezterm, you run something that tries to access the filesystem that contains your photos, calendar and so on, macOS will show a scary dialog that suggests that wezterm is accessing those things.  It isn't: the programs that you instructed your shell to run are doing that.

To make things more annoying, if wezterm doesn't request permissions to do those things by listing them in its manifest, then those permissions will be automatically denied and break those tools.  See https://github.com/wez/wezterm/issues?q=is%3Aissue+permission+camera+is%3Aclosed for some examples.

So wezterm is left in the unfortunate position of having to look like it is asking for permission to do everything when the reality is that it doesn't care and doesn't directly need those permissions.  The same is true of other terminals on macOS.

**Comment by gschwim** [comment]
Thanks for the answers! Explains perfectly.
